lib/rules/selector-anb-no-unmatchable/index.js

```diff
@@ -0,0 +1,121 @@
+'use strict';
+
+const { parse } = require('css-tree');
+const {
+	aNPlusBNotationPseudoClasses,
+	aNPlusBOfSNotationPseudoClasses,
+} = require('../../reference/selectors');
+const isStandardSyntaxRule = require('../../utils/isStandardSyntaxRule');
+
+const report = require('../../utils/report');
+const ruleMessages = require('../../utils/ruleMessages');
+const validateOptions = require('../../utils/validateOptions');
+
+const ruleName = 'selector-anb-no-unmatchable';
+
+const messages = ruleMessages(ruleName, {
+	rejected: (pseudoClass) => `Unexpected unmatchable An+B selector "${pseudoClass}"`,
+	rejectedParseError: (selector) => `Cannot parse selector "${selector}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/selector-anb-no-unmatchable',
+};
+
+function isUnmatchableNth(/** @type {import('css-tree').AnPlusB} */ nth) {
+	const { a, b } = nth;
+
+	if (a !== null && a !== '0' && a !== '-0') {
+		return false;
+	}
+
+	if (b !== null && b !== '0' && b !== '-0') {
+		return false;
+	}
+
+	return true;
+}
+
+/** @type {import('stylelint').Rule} */
+const rule = (primary) => {
+	return (root, result) => {
+		const validOptions = validateOptions(result, ruleName, { actual: primary });
+
+		if (!validOptions) {
+			return;
+		}
+
+		root.walkRules((ruleNode) => {
+			if (!isStandardSyntaxRule(ruleNode)) {
+				return;
+			}
+
+			ruleNode.selectors.forEach((selector, i) => {
+				try {
+					const cssTreeSelector = parse(selector, { context: 'selector', positions: true });
+
+					checkSelector(cssTreeSelector);
```

Comment:
[suggest] Moving `checkSelector(cssTreeSelector)` out of the `try...catch` block seems better to me because the current code may catch an unexpected error by `checkSelector()` .

So, I suggest a refactoring like this:

```js
let cssTreeSelector;

try {
	cssTreeSelector = parse(selector, { context: 'selector', positions: true });
} catch {
	// ...
	return;
}

checkSelector(cssTreeSelector);
```